sdk/tdf_config_test.go

```diff
@@ -95,3 +95,21 @@ func TestWithAssertions(t *testing.T) {
 	assert.Equal(t, id1, cfg.assertions[0].ID)
 	assert.Equal(t, id2, cfg.assertions[1].ID)
 }
+
+func TestWithTargetMode(t *testing.T) {
+	cfg := makeConfig(t, WithTargetMode("0.0.0"))
```

Comment:
Add an empty target mode test, and maybe some without the patch version?

Also, I'd suggest doing this with a table test, just to make the output slightly more debuggable

Reply:
added, right now i have it just default to false if the input is empty "", but if an invalid semver is provided then it will error, does this make sense? or do you think we should also error for ""?

Reply:
I think there might be scenarios where people want to set this value to the default explicitly, if only for convenience, so I think this is good.